### DISCUSSION TITLE
How to tweak html code and the routes?

### DISCUSSION THREAD

**Discussion by naddika** [jlelse/GoBlog#64] [Q&A] [answered]
There's no html layout, is there? There's an XSL for sitemap, I see. What would **the proper way** to edit html code such that this won't mess up with anything else? I'd want to change the main page a little.

 Also, how to tweak the routes? Instead of, let's say, `/posts/YYYY/mm/slug` I want it to become `/articles/YYYY/slug` or `/articles/slug`. In the config, as I've found out, there's no way to do it although there's a key for `menu` with links. 
Is there any way then?

**Comment by jlelse** [comment]
The best way is to use plugins to modify the UI. I know that @kandr3s does it this way. Another way is to edit the ui.go and uiComponents.go files, but that required you to maintain a soft fork.

To customize the routes: To modify the path for new posts, you can change the path template in /settings for the section.

**Reply by naddika** [reply]
Regarding the routes:

I've tried multiple ways of changing "Path template" in /settings but the format of the urls of the posts has remained the untouched -- `/posts/2023/08/slug123`


(1)
```
{{"articles/" .Slug}}
```

(2)
```
{{printf \"/%v/%v\" .Section .Slug}}
```

(3)
```
/articles
```

(4)
```
/articles/slug
```
(5)
```
fdsafdsafdsfds
```


What am I doing incorrectly?

**Reply by naddika** [reply]
![Screenshot_20230818_221342](https://github.com/jlelse/GoBlog/assets/44301454/a5162ef0-e624-4a13-8000-1c97b299467a)

**Reply by jlelse** [reply]
Changing that template only works for future posts. It only effects the "path" parameter that gets assigned to posts. If you want to change a post path, you have to update that parameter on the post.

**Reply by naddika** [reply]
All in all, how to do it in the `/settings`? What precisely will I need to fill into the "path template" field?

**Reply by jlelse** [reply]
There is no setting that applies to existing posts.

**Reply by naddika** [reply]
I already know. I'm asking about the future ones.

**Reply by jlelse** [reply, marked as answer]
One example is

```
{{printf "/%v/%02d/%02d/%v" .Section .Year .Month .Slug}}
```

You can use the following variables:

.Section, .Slug, .Year, .Month, .Day and .BlogPath
